### PR TITLE
PARQUET-1041: Support Arrow's NullArray

### PR DESCRIPTION
**Pull request by xhochy**
This also fixes a segmentation fault on reading a duplicate column.

### CODE REVIEW

**Comment by xhochy** [comment]
cc @crepererum

File: src/parquet/arrow/reader.cc [hunk 1]
```diff
@@ -1262,6 +1260,10 @@ Status PrimitiveImpl::NextBatch(
   }
 
   switch (field_->type()->id()) {
+    case ::arrow::Type::NA:
+      *out = std::make_shared<::arrow::NullArray>(batch_size);
+      return Status::OK();
+      break;
```
Comment (wesm) [inline src/parquet/arrow/reader.cc]: This `break` is unreachable, any reason to keep it?

File: src/parquet/types.h [hunk 2]
```diff
@@ -81,7 +81,8 @@ struct LogicalType {
     INT_64,
     JSON,
     BSON,
-    INTERVAL
+    INTERVAL,
+    NA = 25
```
Comment (wesm) [inline src/parquet/types.h]: Is this from parquet.thrift?

Comment (xhochy) [inline src/parquet/types.h]: Yes

File: src/parquet/arrow/reader.cc (hunk 1 above)
Comment (xhochy) [inline src/parquet/arrow/reader.cc]: Yes, getting rid of compiler warnings.

Comment (wesm) [inline src/parquet/arrow/reader.cc]: gotcha, I hadn't seen that before

**Pull request closed without merging**